src/Compilers/CSharp/Portable/BoundTree/BoundNode.cs

```diff
@@ -158,7 +158,7 @@ public void ResetCompilerGenerated(bool newCompilerGenerated)
             }
         }
 
-        public bool IsSuppressed
+        public virtual bool IsSuppressed
```

Comment:
Did you consider the approach where you just had `BoundExperssionWithNullability` set this in the ctor based on the value of `Expression.IsSuppressed`? That could avoid the use of `virtual` here as long as we were confident that `Expression` would never be set after the ctor.

Reply:
I did, but the constructor is generated.

Reply:
I could add a constructor overload that takes a `bool isSuppressed` flag. Do you think that's better?

Reply:
We could but that would also be eays to miss in code. Should we just remove this ctor from the generator and hand author it? I believe there is some precedence for that. 

---
In reply to: [252452436](https://github.com/dotnet/roslyn/pull/32947#discussion_r252452436) [](ancestors = 252452436)

Reply:
We don't currently have a way of not generating constructors. But I created an overload without `hasErrors`.